app/models/motd_formatter_osc.rb

```diff
@@ -0,0 +1,40 @@
+class MotdFormatterOsc
+
+  attr_reader :content
+
+  # Parse the MOTD in OSC style
+  # See: https://github.com/OSC/ood-dashboard/wiki/Message-of-the-Day
+  #
+  # @param [MotdFile] motd_file an MotdFile object that contains a URI path to a message of the day in OSC format
+  def initialize(motd_file)
+    @content = motd_file.content
+  end
+
+  Message = Struct.new :date, :title, :body do
+    def self.from(str)
+      if str =~ /(\d+[-\/\.]\d+[-\/\.]\d+)\n--- ([ \S ]*)\n(.*)/m
+        Message.new(Date.parse($1), $2, $3)
+      else
+        nil
+      end
+    rescue ArgumentError => e
+      Rails.logger.warn("MOTD message poorly formatted: #{e} => #{e.message}")
+
+      nil
+    end
+  end
+
+  def to_partial_path
+    "dashboard/motd_osc"
+  end
+
+  # since this actually splits the file content into separate messages
+  # we can still use this as it is
+  def messages
+    if content
```

Comment:
motd.content can return nil; then this would be false. `#messages` should probably return an empty array if the content is nil, however